From e36b5b89a42a0a3032d081d423d782b53f4b1d53 Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Fri, 15 Feb 2013 13:10:06 +0000
Subject: [PATCH 3/7] From Alyssa Milburn via
 https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=8335

Make length field unsigned so that negative values fail the bounds check and
throw a regular exception before getting passed to glib (where they cause a
program-ending assert failure instead).

svn path=/trunk/; revision=47672

Conflicts:
	epan/dissectors/packet-mount.c
---
 epan/dissectors/packet-mount.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-mount.c b/epan/dissectors/packet-mount.c
index 2902690..98976bc 100644
--- a/epan/dissectors/packet-mount.c
+++ b/epan/dissectors/packet-mount.c
@@ -170,7 +170,7 @@ dissect_mount_dirpath_call(tvbuff_t *tvb, int offset, packet_info *pinfo,
 			const gchar *host;
 			unsigned char *name;
 			unsigned const char *dir;
-			int len;
+			guint32 len;
 
 			host=ip_to_str(pinfo->dst.data);
 			len=tvb_get_ntohl(tvb, offset);
-- 
1.7.10.4

